Formula/awscli.rb

```diff
@@ -16,22 +27,118 @@ class Awscli < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "014d3f697fd6f83e7056b286ea7c26e78032d21279061df4e151774752b090cd"
   end
 
+  head do
+    url "https://github.com/aws/aws-cli.git", branch: "v2"
+
+    resource "botocore" do
+      url "https://github.com/boto/botocore.git", branch: "v2"
+    end
+  end
+
   depends_on "cmake" => :build
   depends_on "python@3.9"
+  depends_on "six"
 
   uses_from_macos "groff"
+  uses_from_macos "openssl@1.1"
+
+  # Python resources should be updated based on setup.py. One possible way is:
+  # 1. Download source tarball
+  # 2. Remove `botocore` from setup.py
+  # 3. At top of source directory, run `pipgrip . --sort`
+  # 4. Ignore old `botocore` v1 and `six`. Update all other PyPI packages
+
+  resource "awscrt" do
+    url "https://files.pythonhosted.org/packages/9f/e2/5a0b096fb73f2b33ad2cc7392d33d365d3b78b295b21dda5792dd481b38f/awscrt-0.11.24.tar.gz"
+    sha256 "b8aa68bca404bf0085be0570eff5b542d01f7e8e3c0f9b0859abfe5e070162ff"
+  end
 
-  on_linux do
-    depends_on "libyaml"
+  resource "cffi" do
+    url "https://files.pythonhosted.org/packages/2e/92/87bb61538d7e60da8a7ec247dc048f7671afe17016cd0008b3b710012804/cffi-1.14.6.tar.gz"
+    sha256 "c9a875ce9d7fe32887784274dd533c57909b7b1dcadcc128a2ac21331a9765dd"
+  end
+
+  resource "colorama" do
+    url "https://files.pythonhosted.org/packages/82/75/f2a4c0c94c85e2693c229142eb448840fba0f9230111faa889d1f541d12d/colorama-0.4.3.tar.gz"
+    sha256 "e96da0d330793e2cb9485e9ddfd918d456036c7149416295932478192f4436a1"
+  end
+
+  resource "cryptography" do
```

Comment:
Don't we need Rust to build this? Or is this not the version that needs Rust yet?

Reply:
Upstream has pinned `cryptography` to an old version `cryptography>=3.3.2,<3.4.0`.